### PR TITLE
Minhaz

### PR DESCRIPTION
**Pull request by mminhaz93**
- Git commands
- Git ignore

### CODE REVIEW

**Review by mirzabhuiyan — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+.idea/**/mongoSettings.xml
+
+# File-based project format
+*.iws
+
+# IntelliJ
+out/
+
+# mpeltonen/sbt-idea plugin
+.idea_modules/
```
Comment (marziata) [inline .gitignore:57]: Looks good

File: .idea/vcs.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (marziata) [inline .idea/vcs.xml:6]: Maybe we can remove this in future

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/vcs.xml
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
```
Comment (hossainmohammed18) [inline .gitignore:10]: looks good

**Review by marziata — APPROVED** [review]
Looks good

File: README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+git checkout -b BranchName - will take all changed files and create a new branch with it
+            
+git clone url of the repo - will clone the repo to the current directory
+        example : git clone https://github.com/mminhaz93/GitOverview     
+        
+git pull origin BranchName - pulls the latest code of  certain branch
+
+git fetch - pulls the latest branches from git
+
+
```
Comment (Greedy1996) [inline README.md:22]: Look useful

**Review by hossainmohammed18 — APPROVED** [review]
looks good

**Review by Greedy1996 — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+.idea/**/gradle.xml
+.idea/**/libraries
+
+# Gradle and Maven with auto-import
+# When using Gradle or Maven with auto-import, you should exclude module files,
+# since they will be recreated, and may cause churn.  Uncomment if using
+# auto-import.
+# .idea/artifacts
+# .idea/compiler.xml
+# .idea/jarRepositories.xml
```
Comment (FerdousBegum) [inline .gitignore:37]: Looks good

**Review by FerdousBegum — APPROVED** [review]

**Pull request merged**